File: trust-core/src/main/java/trust/core/zcash/ZcashAddressFactory.java
package trust.core.zcash;

import trust.core.blockchain.address.AddressFactory;
import trust.core.entity.address.Address;

public class ZcashAddressFactory implements AddressFactory {

    public static final AddressFactory INSTANCE = new ZcashAddressFactory();


    @Override
    public Address create(String address) {
        return new ZcashAddress(address);
    }

    @Override
    public boolean validate(String address) {
        return ZcashAddress.isAddress(address);
    }

    @Override
    public Address empty() {
        return ZcashAddress.EMPTY;
    }


}

File: trust-core/src/main/java/trust/core/zcash/ZcashAddressFactory.kt
package trust.core.zcash

import trust.core.blockchain.address.AddressFactory
import trust.core.entity.address.Address

class ZcashAddressFactory : AddressFactory {


    override fun create(address: String): Address {
        return ZcashAddress(address)
    }

    override fun validate(address: String): Boolean {
        return ZcashAddress.isAddress(address)
    }

    override fun empty(): Address {
        return ZcashAddress.EMPTY
    }

    companion object {
        val INSTANCE: AddressFactory = ZcashAddressFactory()
    }


}